models/topics.model.js

```diff
@@ -13,3 +13,16 @@ exports.getTopicsData = () => {
     return results.rows;
   });
 };
+
+exports.getArticleDataById = (id) => {
+  return db
+    .query("SELECT * FROM articles WHERE article_id = $1", [id])
+    .then((result) => {
+      if (result.rows[0]) return result.rows[0];
+      return Promise.reject();
+    })
+    .catch((err) => {
+      if (err) return Promise.reject({ code: err.code });
+      return Promise.reject({ status: 404, msg: "article does not exist" });
+    });
```

Comment:
This catch block shouldnt be here - your chain continues in the controller, so the catch should only be there.

Reply:
You've confusingly got two catch blocks atm